runtime/README.md

```diff
@@ -0,0 +1,488 @@
+the Runtime Firmware binary.
+
+## Maibox Commands
+
+All mailbox command codes are little endian.
+
+Table: Mailbox command "result" codes:
+
+| **Name**         | **Value**              | Description
+| -------          | -----                  | -----------
+| `SUCCESS`        | `0x0000_0000`          | Mailbox command succeeded
+| `BAD_VENDOR_SIG` | `0x5653_4947` ("VSIG") | Vendor signature check failed
+| `BAD_OWNER_SIG`  | `0x4F53_4947` ("OSIG") | Owner signature check failed
+| `BAD_SIG`        | `0x4253_4947` ("BSIG") | Generic signature check failure (for crypto offload)
+| `BAD_IMAGE`      | `0x4249_4D47` ("BIMG") | Malformed input image
+| `BAD_CRC`        | `0x4243_5243` ("BCRC") | CRC check failed on input arguments
+
+Relevant Mailbox Registers:
+
+* COMMAND: Command code to execute
+* DLEN: Number of bytes written to mailbox
+
+### CALIPTRA\_FW\_LOAD
+
+The `CALIPTRA_FW_LOAD` command is handled by both ROM and Runtime Firmware.
+
+#### ROM Behavior
+
+On cold boot, ROM will expose the `CALIPTRA_FW_LOAD` mailbox command to accept
+the firmware image that ROM will boot. This includes Manifest, FMC, and Runtime
+firmware.
+
+#### Runtime Firmware Behavior
+
+Caliptra Runtime FW will also expose this mailbox command for loading
+impactless updates. See “Runtime Firmware Updates” for details.
+
+Command Code: `0x4657_4C44` ("FWLD")
+
+Table: `CALIPTRA_FW_LOAD` input arguments
+
+| **Name**  | **Type**      | **Description**
+| --------  | --------      | ---------------
+| data      | u8[...]       | Firmware image to load.
+
+Table: `CALIPTRA_FW_LOAD` output arguments
+
+| **Name** | **Type** | **Description**
+| -------- | -------- | ---------------
+| crc      | u32      | CRC over other output arguments, computed by Caliptra. Little endian.
+| result   | u32      | Result code. Little endian.
+
+### GET\_IDEV\_CSR
+
+ROM exposes a command to get a self-signed IDEVID CSR.
+GET\_IDEV\_CSR is not exposed by runtime firmware.
+
+Command Code: `0x4944_4556` ("IDEV")
+
+`GET_IDEV_CSR` takes no input arguments.
+
+* When invoked via JTAG, ROM will write the CSR to the mailbox so it can
+  be read out over JTAG
+
+Table: `GET_IDEV_CSR` output arguments
+
+| **Name**  | **Type**      | **Description**
+| --------  | --------      | ---------------
+| data      | u8[...]       | DER-encoded IDevID CSR
+
+### GET\_LDEV\_CERT
+
+ROM exposes a command to get a self-signed LDevID Certificate signed by IDevID.
+GET\_LDEV\_CERT is not exposed by runtime firmware.
+
+Command Code: `0x4C44_4556` ("LDEV")
+
+`GET_LDEV_CERT` takes no input arguments.
+
+* When invoked via JTAG, ROM will write the certificate to the mailbox so it can
+  be read out over JTAG
+
+Table: `GET_LDEV_CERT` output arguments
+
+| **Name**  | **Type**      | **Description**
+| --------  | --------      | ---------------
+| data      | u8[...]       | DER-encoded LDevID Certificate
+
+### ECDSA384\_SIGNATURE\_VERIFY
```

Comment:
need an open on whether this should take the hash from the SHA block in accordance with https://github.com/chipsalliance/Caliptra/blob/main/doc/Caliptra.md#soc-fw-authentication-flows

Reply:
Added an open at the bottom